### ISSUE TITLE
Failed with output after run trainer script

### ISSUE THREAD

**Issue by iqfareez** [KOE-Wayfind/AR-server#3]
Error running 

```shell
python my_hloc.py
```

Console:
```shell
[2024/03/13 01:10:05 hloc INFO] Importing features into the database...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70/70 [00:00<00:00, 1990.95it/s]
[2024/03/13 01:10:05 hloc INFO] Importing matches into the database...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2415/2415 [00:01<00:00, 1475.46it/s]
[2024/03/13 01:10:07 hloc INFO] Performing geometric verification of the matches...
[2024/03/13 01:10:07 hloc ERROR] Failed with output:

Traceback (most recent call last):
  File "/workspaces/AR-server/my_hloc.py", line 109, in <module>
    main()
  File "/workspaces/AR-server/my_hloc.py", line 62, in main
    model = reconstruction.main(sfm_dir, images, sfm_pairs, features, matches, image_list=references)
  File "/workspaces/AR-server/Hierarchical-Localization/hloc/reconstruction.py", line 121, in main
    estimation_and_geometric_verification(database, pairs, verbose)
  File "/workspaces/AR-server/Hierarchical-Localization/hloc/triangulation.py", line 109, in estimation_and_geometric_verification
    pycolmap.verify_matches(
TypeError: verify_matches(): incompatible function arguments. The following argument types are supported:
    1. (database_path: str, pairs_path: str, options: pycolmap.TwoViewGeometryOptions = <pycolmap.TwoViewGeometryOptions object at 0x7fe874cb6530>) -> None

Invoked with: PosixPath('outputs/demo/sfm/database.db'), PosixPath('outputs/demo/pairs-sfm.txt'); kwargs: max_num_trials=20000, min_inlier_ratio=0.1
```

Looks like dependency issue

**Comment by iqfareez** [comment]
Fixed by constraining pycolmap to version `0.4.0`.

Running `pip freeze`:
```txt
anyio==4.3.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==2.4.1
async-lru==2.0.4
attrs==23.2.0
Babel==2.14.0
beautifulsoup4==4.12.3
bleach==6.1.0
blinker==1.7.0
certifi==2024.2.2
cffi==1.16.0
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
comm==0.2.1
contourpy==1.2.0
cycler==0.12.1
debugpy==1.8.1
decorator==5.1.1
defusedxml==0.7.1
exceptiongroup==1.2.0
executing==2.0.1
fastjsonschema==2.19.1
filelock==3.13.1
Flask==3.0.2
fonttools==4.49.0
fqdn==1.5.1
fsspec==2024.2.0
gdown==5.1.0
gitdb==4.0.11
GitPython==3.1.42
h11==0.14.0
h5py==3.10.0
-e git+https://github.com/KOE-Wayfind/Hierarchical-Localization@1a30076c18bb8723799f20c4ebc30ff6b7579ba1#egg=hloc
httpcore==1.0.4
httpx==0.27.0
idna==3.6
ipykernel==6.29.3
ipython==8.22.2
isoduration==20.11.0
itsdangerous==2.1.2
jedi==0.19.1
Jinja2==3.1.3
joblib==1.3.2
json5==0.9.20
jsonpointer==2.4
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
jupyter-events==0.9.0
jupyter-lsp==2.2.4
jupyter-server-mathjax==0.2.6
jupyter_client==8.6.0
jupyter_core==5.7.1
jupyter_server==2.13.0
jupyter_server_terminals==0.5.2
jupyterlab==4.1.3
jupyterlab_git==0.50.0
jupyterlab_pygments==0.3.0
jupyterlab_server==2.25.3
kiwisolver==1.4.5
kornia==0.7.1
MarkupSafe==2.1.5
matplotlib==3.8.3
matplotlib-inline==0.1.6
mistune==3.0.2
mpmath==1.3.0
nbclient==0.9.0
nbconvert==7.16.2
nbdime==4.0.1
nbformat==5.9.2
nest-asyncio==1.6.0
networkx==3.2.1
notebook_shim==0.2.4
numpy==1.26.4
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu12==2.19.3
nvidia-nvjitlink-cu12==12.3.101
nvidia-nvtx-cu12==12.1.105
opencv-python==4.9.0.80
overrides==7.7.0
packaging==23.2
pandas==2.2.1
pandocfilters==1.5.1
parso==0.8.3
pexpect==4.9.0
pillow==10.2.0
platformdirs==4.2.0
plotly==5.19.0
prometheus_client==0.20.0
prompt-toolkit==3.0.43
psutil==5.9.8
ptyprocess==0.7.0
pure-eval==0.2.2
pycolmap==0.4.0
pycparser==2.21
Pygments==2.17.2
pyparsing==3.1.1
PySocks==1.7.1
python-dateutil==2.9.0.post0
python-json-logger==2.0.7
pytz==2024.1
PyYAML==6.0.1
pyzmq==25.1.2
referencing==0.33.0
requests==2.31.0
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rpds-py==0.18.0
scikit-learn==1.4.1.post1
scipy==1.12.0
seaborn==0.13.2
Send2Trash==1.8.2
six==1.16.0
smmap==5.0.1
sniffio==1.3.1
soupsieve==2.5
stack-data==0.6.3
sympy==1.12
tenacity==8.2.3
terminado==0.18.0
threadpoolctl==3.3.0
tinycss2==1.2.1
tomli==2.0.1
torch==2.2.1
torchvision==0.17.1
tornado==6.4
tqdm==4.66.2
traitlets==5.14.1
triton==2.2.0
types-python-dateutil==2.8.19.20240106
typing_extensions==4.10.0
tzdata==2024.1
uri-template==1.3.0
urllib3==2.0.7
wcwidth==0.2.13
webcolors==1.13
webencodings==0.5.1
websocket-client==1.7.0
Werkzeug==3.0.1
```

**Comment by iqfareez** [comment]
Fixed
